WordPress/src/main/java/org/wordpress/android/ui/posts/reactnative/ReactNativeRequestHandler.kt

```diff
@@ -0,0 +1,83 @@
+package org.wordpress.android.ui.posts.reactnative
+
+import androidx.core.util.Consumer
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.store.ReactNativeFetchResponse
+import org.wordpress.android.fluxc.store.ReactNativeFetchResponse.Error
+import org.wordpress.android.fluxc.store.ReactNativeFetchResponse.Success
+import org.wordpress.android.fluxc.store.ReactNativeStore
+import org.wordpress.android.modules.BG_THREAD
+import javax.inject.Inject
+import javax.inject.Named
+
+class ReactNativeRequestHandler @Inject constructor(
+    private val reactNativeStore: ReactNativeStore,
+    private val urlUtil: ReactNativeUrlUtil,
+    @Named(BG_THREAD) private val bgDispatcher: CoroutineDispatcher
+) : CoroutineScope {
+    override val coroutineContext = bgDispatcher + Job()
+
+    fun performGetRequest(
+        pathWithParams: String,
+        mSite: SiteModel,
+        onSuccess: Consumer<String>,
+        onError: Consumer<String>
+    ) {
+        launch {
+            if (mSite.isUsingWpComRestApi) {
+                performGetRequestForWPComSite(pathWithParams, mSite.siteId, onSuccess::accept, onError::accept)
+            } else {
+                performGetRequestForSelfHostedSite(pathWithParams, mSite.url, onSuccess::accept, onError::accept)
+            }
+        }
+    }
+
+    /**
+     * A given instance of this class may not be used after [destroy] is called because:
+     *     (1) this class's coroutineContext has a single job instance that is created on initialization;
+     *     (2) calling `destroy()` cancels that job; and
+     *     (3) jobs cannot be reused once cancelled.
+     */
+    fun destroy() {
+        coroutineContext[Job]!!.cancel()
```

Comment:
I also toyed with the idea of making this something like:

```
coroutineContext[Job]?.apply {
  cancel()
} ?: AppLog.e(EDITOR, "Scope could not be cancelled because it does not have a job. This should never happen.")
```
I decided against this approach since this class creates a `CoroutineContext` with a `Job` on initialization, but let me know if anyone prefers this (or some other) approach.

Reply:
I usually just keep the instance of the `job` and cancel it directly but I think this approach is fine 👍